### PR TITLE
NEW : add default warehouse for dispatch

### PR DESCRIPTION
**Pull request by altairis-tof**

### CODE REVIEW

**Comment by fappels** [comment]
Product has a field 'fk_default_warehouse', better use this value as default of the select.
Also please use fk_default_warehouse iso fk_entrepot_default

**Comment by altairis-tof** [comment]
no default value here because it's a default warehouse for different products !

**Comment by fappels** [comment]
Ok, thanks didn't see this. Maybe one small remark, seems product default warehouse is already implemented and this will clear default product warehouse.

**Comment by altairis-tof** [comment]
it will clear product default warehouse only if one value is selected in this new field.
I think that if someone use the product default warehouse feature, he will not use this new feature.
But for those who receive big orders in their warehouse, it's very useful to select their warehouse only one and not for each line of the supplier order ;-)

**Comment by altatof** [comment]
je vais essayer en français des fois que mon anglais soit pourri... j'ai une dizaine de PR en attente depuis début mars... comment voulez vous que les devs aient envie de s'investir si leur boulot n'est pas valorisé ? En + on est obligé de faire du spécif chez les clients en attendant le bon vouloir d'un tel ou d'un tel... fais chier !

**Comment by wdammak** [comment]
@eldy : what's the problem with this PR?

**Comment by atm-maxime** [comment]
Hmm... Maybe continuous integration failing ?

**Comment by altairis-tof** [comment]
and ? what to do ?

**Comment by atm-maxime** [comment]
Go on Travis CI to check the errors.
I checked here : https://travis-ci.org/Dolibarr/dolibarr/jobs/526839649#L1306

There is only one error in your file, a space missing. I suggest you pull the develop branch first to remove other errors that should not be present in develop branch now.

Also try to use comments in English, not in French ;)

**Comment by altairis-tof** [comment]
a space missing...

**Comment by altairis-tof** [comment]
i mean that if you stop PR because of missing spaces, there will not be PR accepted anymore because missing spaces are everywhere in dolibarr core code ! 
i would prefer focus on duplicate code or functional diseases (like refreshing the page after choosing a column in a list) than to this kind of ... stuff to stay polite ;-)

**Comment by altairis-tof** [comment]
but it is my own view... let's respect standards !

**Comment by atm-maxime** [comment]
I think you'll agree that having coding rules is a good way to have every contributor coding the same way. So it is easier for everyone to read/understand code made by someone else, and to contribute. Coding rules has not been invented by Dolibarr, we're following PSR2.

Travis only checks your added lines so your PR will be blocked only if the new code doesn't respect rules. To deal with errors already present we use scrutinizer tool.

Travis is in place for some time now and the number of PRs didn't stop increasing. So Travis is used as a tool by @eldy to sort PRs that he can check and merge and the ones that he can pass on because coding rules are not respected.

I am grumbling too when a piece of code I add is pointed at by Travis, but then I remember that I don't just code for myself but I am a part of a community and we all need to work together.

**Comment by altairis-tof** [comment]
missing space fixed :) you can merge ;-)

**Comment by altatof** [comment]
image
what should we do now ? @atm-maxime @eldy

**Comment by altatof** [comment]
better and better :

image

and when i click on "i can make one for you" and give my github credentials (!!!) to stickler : 

image

**Comment by altatof** [comment]
i will succeed...

**Comment by altairis-tof** [comment]
yes it's ok now !!! @eldy @atm-maxime

File: htdocs/fourn/commande/dispatch.php [hunk 1]
```diff
@@ -236,6 +239,7 @@
 			$prod = "product_" . $reg[1] . '_' . $reg[2];
 			$qty = "qty_" . $reg[1] . '_' . $reg[2];
 			$ent = "entrepot_" . $reg[1] . '_' . $reg[2];
+			if(empty($ent)) $ent = $fk_default_warehouse;
```
Comment (eldy) [inline htdocs/fourn/commande/dispatch.php]: $ent will never be empty

**Comment by altairis-tof** [comment]
ready to merge

**Pull request merged**